### DISCUSSION TITLE
Get as custom type

### DISCUSSION THREAD

**Discussion by freerig** [automerge/automerge#936] [Q&A]
I want to do something like this :
```rs
let value = doc.get(automerge::ROOT, "list")
// Some magic stuf
let final_value: Vec<VeryCustomStruct> = ... // So final_value is the same as value but with the right types
```

**Comment by alexjg** [comment]
Have you had a look at [autosurgeon](https://docs.rs/autosurgeon/latest/autosurgeon/)?

**Reply by freerig** [reply]
Yes, I did, but I was wondering if there wasn't an easier way to do it. So I have to use hydrate, right ? Do you know a repo that uses autosurgeon to have an exemple ?
Thanks
